module/planning/PlanKick/src/PlanKick.cpp

```diff
@@ -0,0 +1,93 @@
+#include "PlanKick.hpp"
+
+#include <string>
+
+#include "extension/Behaviour.hpp"
+#include "extension/Configuration.hpp"
+
+#include "message/localisation/FilteredBall.hpp"
+#include "message/planning/KickTo.hpp"
+#include "message/skill/Kick.hpp"
+
+#include "utility/input/LimbID.hpp"
+#include "utility/support/yaml_expression.hpp"
+
+namespace module::planning {
+
+    using extension::Configuration;
+    using message::localisation::FilteredBall;
+    using message::planning::KickTo;
+    using message::skill::Kick;
+    using utility::input::LimbID;
+    using utility::support::Expression;
+
+    PlanKick::PlanKick(std::unique_ptr<NUClear::Environment> environment) : BehaviourReactor(std::move(environment)) {
+
+        on<Configuration>("PlanKick.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file PlanKick.yaml
+            this->log_level             = config["log_level"].as<NUClear::LogLevel>();
+            cfg.ball_timeout_threshold  = config["timeout_threshold"].as<float>();
+            cfg.ball_distance_threshold = config["ball_distance_threshold"].as<float>();
+            cfg.ball_angle_threshold    = config["ball_angle_threshold"].as<float>();
+            cfg.target_angle_threshold  = config["target_angle_threshold"].as<float>();
+            cfg.kick_leg                = config["kick_leg"].as<std::string>();
+        });
+
+        on<Provide<KickTo>, Trigger<FilteredBall>, Needs<Kick>>().then(
```

Comment:
The planner should not `Needs<Kick>` the idea would be that it wants to kick and tries by emitting a task. If it needs kick you can't even plan to kick unless the walk isn't running.